packages/@aws-cdk/aws-redshift/lib/cluster-parameter-change-reboot-handler/index.ts

```diff
@@ -0,0 +1,53 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { Redshift } from 'aws-sdk';
+
+const redshift = new Redshift();
+
+export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent): Promise<void> {
+  if (event.RequestType !== 'Delete') {
+    return rebootClusterIfRequired(event.ResourceProperties?.ClusterId, event.ResourceProperties?.ParameterGroupName);
+  } else {
+    return;
+  }
+}
+
+async function rebootClusterIfRequired(clusterId: string, parameterGroupName: string): Promise<void> {
+  return executeActionForStatus(await getApplyStatus());
+
+  // https://docs.aws.amazon.com/redshift/latest/APIReference/API_ClusterParameterStatus.html
+  async function executeActionForStatus(status: string, retryDurationMs?: number): Promise<void> {
+    await sleep(retryDurationMs ?? 0);
+    if (['pending-reboot', 'apply-deferred', 'apply-error', 'unknown-error'].includes(status)) {
+      try {
+        await redshift.rebootCluster({ ClusterIdentifier: clusterId }).promise();
+      } catch (err) {
+        if ((<any>err).code === 'InvalidClusterState') {
+          return await executeActionForStatus(status, 30000);
```

Comment:
I thought about doing some sort of backoff here, but found checking every `30` seconds to be sufficient in my testing.